buildtest/menu/build.py

```diff
@@ -74,19 +74,57 @@ def discover_configs(config_file):
         logger.error(msg)
         sys.exit(msg)
 
+    # return all configs by resolving path, this gets the real canonical path and address shell expansion and user expansion
+    config_files = [resolve_path(config) for config in config_files]
+
     logger.info(f"Found the following config files: {config_files}")
     return config_files
 
 
+def include_file(file_path, white_list_patterns):
+    """Check if file is included based on OR regular expression. If no white_list_patterns
+       provided method will return ``True``. Otherwise it will return the list of files that don't
+       match the regular expression
+
+       Parameters:
+
+       :param file_path: file path to run regular expression upon
+       :type file_path: str, required
+       :param white_list_patterns: the exclude list provided on command line option
+       :type white_list_patterns: list, required
+       :return: Returns True or a list of files that don't match regular expression
+       :rtype: bool or str
+    """
+
+    logger = logging.getLogger(__name__)
+
+    if not white_list_patterns:
+        logger.debug("No white list patterns provided, returning True")
+        return True
+
+    logger.debug(f"white list pattern before resolving paths: {white_list_patterns}")
+
+    white_list_patterns = [resolve_path(path) for path in white_list_patterns]
+
+    logger.debug(f"white list pattern after resolving paths: {white_list_patterns}")
+
+    regexp = "(%s)" % "|".join(white_list_patterns)
+    logger.debug(f"Applying Regular Expression Search: {regexp} to file: {file_path}")
```

Comment:
This is probably one too many lines for the logger - is there a lower level than debug (e.g., verbose) for at least the last line?

Reply:
DEBUG is the lowest level i believe according to https://docs.python.org/3/howto/logging.html We could either remove the line if it's too much detail or we can set default debug setting to INFO. The default is ``WARNING``.

I figured it could be useful for users to troubleshoot how exclusion is happening if they have information on the regex command.

Reply:
ah okay, then okay to leave it. I think this is why I implemented a custom logger for most of my libraries that has also logger.verbose().